### ISSUE TITLE
predict_proba returns proportions of training set

### ISSUE THREAD

**Issue by agombert** [huggingface/setfit#369]
Hello, 

First thanks for the repo and tuto, really great work !

I have a problem with the training and outputs of probabilities. [Here](https://colab.research.google.com/drive/1YOD9W-mmTVbdJSMOFsJeoO21ewCxkwip#scrollTo=6_YDxWVOLdXN) is the collab I'm using right now. 

I try to train on tweets to get a climate change model on spanish, binary classification. I have 45 tweets examples of each class. And when I train the model with 4, 8 or 44 examples in each class, the probability is always 0.5 in the output. I must do something wrong. I don't understand. 

This is examples of code I have:

**Examples**

```python
positive_examples = [
    'Me preocupa el futuro que estamos dejando para las generaciones futuras. Es hora de actuar con responsabilidad y frenar el cambio climático.',
    '¡No tires eso! Podría ser reciclable. Cada uno de nosotros puede hacer nuestra parte para reducir la cantidad de basura que generamos',
    'Este evento será replicado en todos los estados del país con el objetivo de promover el cuidado del medio ambiente a través del reciclaje.',
    'El agua es un recurso limitado, expertos explican cómo cuidarlo - La Hora https://t.co/ImToMkH67n',
    'La Japay evita el desperdicio de importante cantidad de agua https://t.co/5W88BZdALp https://t.co/Pn9yqApXJZ',
    'Conoce el impacto climático en la sequía y la desertificación https://t.co/ygGPnIzTwG',
    'Australia usa el movimiento del océano para generar energía renovable y desalinizar agua simultáneamente',
    '#Internacionales Energías renovables dan empleo a casi 13 millones de personas https://t.co/GcJeEyrCtJ'
]

negative_examples = [
    'Les comparto mi entrevista del día de hoy con Eduardo Rodriguez acerca de las últimas medidas del BCV en el crédito bancario, la coyuntura del sistema bancario y sus impactos en la economía venezolana',
    '@claudioea123 @alkarrenos @wallstwolverine Si tuviesen la misma cantidad, podrían hacerlo. Pero producen menos (o pierden más de lo que producen de camino al extranjero). Y no porque sus Estados sean más grandes, porque casi ninguno lo es. Y, hasta donde sé, hay menos trabas a la empresa privada que en países de la UE.',
    'Sobre el grupo de niños que estaba desaparecido en las Tejerías, que el día de la tragedia estaban en una fiesta y que aparecieron todos vivos, esto es lo que se está comentando. ¿Qué opinan? #LasTejerias https://t.co/WdCoT1DFoG',
    'https://t.co/KWy5Y3Jkvk Informe sobre seguridad: Maduro entregó en comodato partes del territorio para su explotación. Sepa más, lea los Breves en Crímenes sin Castigo.',
    'Hay una justicia más allá de la terrenal y pueden estar seguros de que el infierno es espantoso,todos los comunistas están en el infierno pagando sus crímenes atroces y se quemaran en el fuego por los siglos de los siglos,por eso el diablo viste de rojo y todos los que le siguen',
    'Por favor para reportar que NO HAY internet desde el miercoles 04/09/2019 en la empresa Papelería Moderna CA J-300742571 Táchira San Cristóbal Parroquia La Concordia Barrio El Carmen Zona Comercial Metropolitana Carrera 10 con Calle 3 0276-3474742 @salaprensaCantv',
    'Olivares: Sebin cometió un error en medio de las torturas https://t.co/Kzq8ofVOkk  https://t.co/W1VtVHzi7H',
    'Los trabajadores de los servicios de inteligencia europeos: https://t.co/xddgaChx5Q'
]

```

**Training**

```python
from datasets import load_dataset, Dataset
from sentence_transformers.losses import CosineSimilarityLoss

from setfit import SetFitModel, SetFitTrainer

model = SetFitModel.from_pretrained('sentence-transformers/distiluse-base-multilingual-cased-v1')

m = 8
texts_train = negative_examples + positive_examples
train_ds = Dataset.from_dict({
    "text": texts_train, 
    "label": [0] * m + [1] * (m),
    "label_text": ['random'] * m + ['environment'] * (m)
                        })

# Create trainer
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_ds,
    loss_class=CosineSimilarityLoss,
    batch_size=16,
    learning_rate=1e-3,
    num_iterations=20, # Number of text pairs to generate for contrastive learning
    num_epochs=1 # Number of epochs to use for contrastive learning
)

trainer.train()
```

**Inference**

```python
model.predict_proba(['cambio climático', 'me gusta la ecología', 'problema de sostenibilidad'])
```

Output:
```python
tensor([[0.5000, 0.5000],
        [0.5000, 0.5000],
        [0.5000, 0.5000]], dtype=torch.float64)
```

I don't understand the problem

**Comment by agombert** [comment]
mmm, looks like reducing learning rate to `1e-4`and increase batch size to `32` helps. Do you have any study on hyper-parameters ? I don't remember reading something on that on the paper.

**Comment by tomaarsen** [comment]
Glad to hear you got it working well!

I'll close this for now :)

- Tom Aarsen
